### PR TITLE
Check for assignment type/format when retrieving assignments.

### PR DESCRIPTION
**Pull request by dsjen**
@clintonb, this merges both of our changes for the assignment structure fix.  The most recent update are fixes to broken acceptance tests in `acceptance_tests/test_course_performance.py`.

### CODE REVIEW

**Comment by clintonb** [comment]
@dsjen If you want to avoid rebasing and conflicts, feel free to copy the code from clintonb/structure-parse-fix. It has already been rebased.

**Comment by dsjen** [comment]
@clintonb thanks, I'll take a look at it--seems to just bet the Makefile.

File: acceptance_tests/test_course_performance.py [hunk 1]
```diff
@@ -304,7 +310,7 @@ def _test_table(self):
             for col in columns:
                 actual.append(col.text)
 
-            expected = [answer[value_field]]
+            expected = [answer[value_field] if answer[value_field] else u'(empty)']
```
Comment (clintonb) [inline acceptance_tests/test_course_performance.py:313]: What is this?

File: acceptance_tests/test_course_performance.py [hunk 2]
```diff
@@ -70,6 +70,12 @@ def _get_assignments(self, assignment_type=None):
 
             for problem in assignment['problems']:
                 submission_entry = problems.get(problem['id'], None)
+
+                problem.update({
```
Comment (clintonb) [inline acceptance_tests/test_course_performance.py:74]: Why is this added here?

File: acceptance_tests/test_course_performance.py (hunk 1 above)
Comment (dsjen) [inline acceptance_tests/test_course_performance.py:313]: The answer can be `None`, and in this case, we display "(empty)" in the data table.

Comment (clintonb) [inline acceptance_tests/test_course_performance.py:313]: In the future, please separate features/additions that aren't directly related into multiple PRs.

File: acceptance_tests/test_course_performance.py (hunk 2 above)
Comment (dsjen) [inline acceptance_tests/test_course_performance.py:74]: Line 242 & 243 (`self.format_number(problem['correct_submissions']),`) fails on particular test problems with no submissions.

Comment (clintonb) [inline acceptance_tests/test_course_performance.py:74]: Okay.

**Comment by clintonb** [comment]
@dsjen Ping me when conflicts have been resolved, and I'll finish my review.

**Comment by dsjen** [comment]
This branch is successfully rebased--not sure why Github isn't showing it as merge safe.  I can manually push when ready.

**Comment by clintonb** [comment]
@dsjen This branch was not rebased on the latest version of master. My attempts to rebase locally fail with many merge conflicts.

**Comment by dsjen** [comment]
Thanks, rebased.

**Comment by clintonb** [comment]
:+1:
